rekuho/src/lib/Portion.svelte

```diff
@@ -0,0 +1,177 @@
+<!--
+    Portion, may be simply the Editor
+    This window contains the visible and editable text.
+-->
+<script lang="ts">
+  import type { Theme } from "./Theme"
+  import LinesView, { lines, isAlpha, insertAt } from "./LinesView.svelte"
+  import type { Cursor, Position } from "./Cursors.svelte"
+  import CursorsLayer, { cursors, cursorUpdate, cursorMove } from "./Cursors.svelte"
+  import Gutter from "./Gutter.svelte"
+
+  export let theme: Theme
+  const gutter_width = 50 // maybe should be part of theme, minimum value?
+
+  export let height: number
+  export let width: number
+
+  let view: Element
+  let input: HTMLTextAreaElement
+  let container: Element
+
+  // TODO: Get proper input from backend
+  cursors.update((_) => [{ pos: [2, 1] }])
+  lines.update((_) => [
+    ...new Array(10).fill(""),
+    ..."funky\nbanana\nt0wn".split("\n"),
+    "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+    ...new Array(20).fill("a"),
+  ])
+
+  // let portion_start_line: number = 0
+
+  const getCharacterWidth = (font: string): number | null => {
+    const canvas = new OffscreenCanvas(0, 0)
+    const context = canvas.getContext("2d") as OffscreenCanvasRenderingContext2D | null
+    if (context) {
+      context.font = font
+    }
+    // FIXME: Elkowar mentioned: "with a font of **ZERO WIDTH X-CHARACTERS**" this breaks.
+    // It does. ture. (typo intended)
+    return context?.measureText("X").width || null
+  }
+
+  // TODO: Separate into linear_algebra.ts
+  $: view_rect = container && container.getBoundingClientRect()
```

Comment:
What's this?

Reply:
https://svelte.dev/tutorial/reactive-declarations